### DISCUSSION TITLE
Remediation task state is successful even with Non-compliant state

### DISCUSSION THREAD

**Discussion by uaghani** [gettek/terraform-azurerm-policy-as-code#89] [Q&A] [answered]
This module works amazingly and eases the pain of deploying custom policies and merging their parameters.

For me, the remediation task works fine (goes through Evaluation, Running, Success) when deployed on Subscription scope but,  always shows successful when deployed on Management Group (with multiple Subscriptions below). 
Compliance shows the data as non-compliant but the task stays in a 'Successful' state.
Any assistance would be appreciated! 

Thank you

**Comment by gettek** [comment, marked as answer]
Hi @uaghani, I'd need to see some code in order to replicate this issue

Note that [re-evaluating compliance](https://github.com/gettek/terraform-azurerm-policy-as-code#%EF%B8%8Fon-demand-evaluation-scan) during remediation tasks only applies at subscription scope and below, this may explain the issue you are facing.

**Reply by uaghani** [reply]
Thanks a ton. This was the issue I was facing. 
Workaround: Rerunning the TF Apply to recreate remediation tasks as well as with 're_evaluate_compliance = true' remediated the resources and updated compliance status along with the deployment status and matched resources to remediate.
